### ISSUE TITLE
Enhancement: new intrinsic -- look up code hash

### ISSUE THREAD

**Issue by cc32d9** [EOSIO/eos#9851]
When a contract is making an outgoing transfer, the recipient account might run a contract that may reject the whole transaction, or abuse the CPU resource. 

New intrinsic: `get_code_hash`  would allow a contract decide if it wants to interact with a particular account, 

Also in multi-contract projects, the contract would be able to verify their neighbor compatibility.

**Comment by aclark-b1** [comment]
Would this meet the use case you're describing?
https://github.com/EOSIO/eos/issues/6168

**Comment by cc32d9** [comment]
> Would this meet the use case you're describing?
> #6168

yes, thanks, it describes it well. Any information when/if it's planned to be implemented?

**Comment by aclark-b1** [comment]
It's too late to get it into 2.1 at this point, but we'll try to target it for the next version which is still in the very early planning stages.

**Comment by goodengineer** [comment]
Hi,
I beg your pardon, can you please shed a bit light as for sha256 correctness and utility in hashing code and account day, maybe I don't understand the hashing in this specific context, it will make it much more clear and understandable if you answer the next questions:
a) Can different smart contract / codes have the same hash digest? SHA256 is not perfect nor injection and so how many collisions per code length Lc can be?
b) How the hash digest of code used afterwards, if comparison operations only, then which digest with which digest is going to be compared and for what?
c) Are you going to compress code before hashing it? If yes, than which compression methods to use for maximal accuracy?

**Comment by conr2d** [comment]
@aclark-b1 I made the PR #7835 which provides this feature long time ago, but not merged then, now it becomes so outdated. If you consider merging my PR, I will update it, but if b1 devs want to implement by themselves, it would be good to let me know.
